packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,373 @@
+    return instance._unregister(widgetSensitivity);
+  }
+
+  Future<void> _unregister(ContentSensitivity widgetSensitivity) async {
+    assert(
+      _contentSenstivityIsSupported != null,
+      'SensitiveContentHost.register must be called before SensitiveContentHost.unregister',
+    );
+
+    if (!_contentSenstivityIsSupported!) {
+      // Setting content sensitivity is not supported on this device.
+      return;
+    }
+
+    // Check current calculated content sensitivity. Use this to determine whether or not
+    // a new content sensitivity needs to be set based on which sensitiivty needs to be
+    // restored to accurately reflect the SensitiveContent widgets in the tree.
+    final ContentSensitivity contentSensitivityBasedOnWidgetCountsBeforeUnregister =
+        _contentSensitivitySetting.contentSensitivityBasedOnWidgetCounts!;
+
+    // Update the content sensitivity estting to account for removing a SensitiveContent
+    // widget with sensitivity widgetSensitivity from the tree.
+    _contentSensitivitySetting.removeWidgetWithContentSensitivity(widgetSensitivity);
+
+    if (!_contentSensitivitySetting.hasWidgets) {
+      // Restore fallback content sensitivity setting if there are no more SensitiveContent
+      // widgets in the tree.
+      if (contentSensitivityBasedOnWidgetCountsBeforeUnregister ==
+          _fallbackContentSensitivitySetting) {
+        return;
+      }
+
+      try {
+        await _sensitiveContentService.setContentSensitivity(_fallbackContentSensitivitySetting!);
+      } catch (e) {
+        FlutterError.reportError(
+          FlutterErrorDetails(
+            exception: FlutterError(
+              'Attempted to set $_fallbackContentSensitivitySetting sensitivity failed: $e}',
+            ),
+            library: 'widget library',
+            stack: StackTrace.current,
+          ),
+        );
+      }
+      return;
+    }
+
+    // Determine if another sensitivity needs to be restored. The null check should be
+    // safe because contentSensitivityBasedOnWidgetCounts should always be non-null as long
+    // as there are still SensitiveContent widgets in the tree.
+    final ContentSensitivity contentSensitivityToRestore =
+        _contentSensitivitySetting.contentSensitivityBasedOnWidgetCounts!;
+    if (contentSensitivityToRestore != contentSensitivityBasedOnWidgetCountsBeforeUnregister) {
+      // Set content sensitivity as contentSensitivityToRestore.
+      try {
+        await _sensitiveContentService.setContentSensitivity(contentSensitivityToRestore);
+      } catch (e) {
+        FlutterError.reportError(
+          FlutterErrorDetails(
+            exception: FlutterError(
+              'Attempted to set $_fallbackContentSensitivitySetting sensitivity failed: $e}',
+            ),
+            library: 'widget library',
+            stack: StackTrace.current,
+          ),
+        );
+      }
+    }
+  }
+}
```

Comment:
This class looks good now, thanks for making the changes. Management of async stuff and error handling all make sense to me.